src/datasets/arrow_dataset.py

```diff
@@ -468,6 +468,7 @@ def save_to_disk(self, dataset_path: str, fs=None):
         with tempfile.TemporaryDirectory() as tmp_dir:
             temp_dataset_path = Path(tmp_dir).joinpath(dataset_path)
             os.makedirs(temp_dataset_path, exist_ok=True)
+            os.makedirs(dataset_path, exist_ok=True)
```

Comment:
```suggestion
            fs.makedirs(dataset_path, exist_ok=True)
```
dataset_path could be a remote filesystem